src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -232,12 +233,65 @@ trait GoldTransforms extends SparkSessionWrapper {
       .otherwise(lit(null).cast("boolean"))
   }
 
+  /**
+   * Temporary nastiness necessary to handle edge cases of states between runs. This will go away in 0.5.1
+   * @param df
+   * @param timeFilter
+   * @return
+   */
+  private def getEventStates(df: DataFrame, timeFilter: Column): DataFrame = {
```

Comment:
Super nasty, but it works...continuing to test but so far it looks good. This is just a quick fix to get a workable solution, this entire logic will be enhanced in next version.